packages/fxa-auth-server/lib/routes/password.js

```diff
@@ -458,6 +490,101 @@ module.exports = function (
         }
       },
     },
+    {
+      // This endpoint will eventually replace '/password/forgot/send_code'
+      // below.  That is also the reason for the similarity between them.
+      method: 'POST',
+      path: '/password/forgot/send_otp',
+      options: {
+        ...PASSWORD_DOCS.PASSWORD_FORGOT_SEND_OTP_POST,
+        validate: {
+          query: isA.object({
+            service: validators.service.description(DESCRIPTION.serviceRP),
+            keys: isA.boolean().optional(),
+          }),
+          payload: isA.object({
+            email: validators
+              .email()
+              .required()
+              .description(DESCRIPTION.emailRecovery),
+            service: validators.service.description(DESCRIPTION.serviceRP),
+            metricsContext: METRICS_CONTEXT_SCHEMA,
+          }),
+        },
+      },
+      handler: async function (request) {
+        if (!config.passwordForgotOtp.enabled) {
+          throw error.featureNotEnabled();
+        }
+
+        log.begin('Password.forgotOtp', request);
+        await request.emitMetricsEvent('password.forgot.send_otp.start');
+        const email = request.payload.email;
+
+        // TODO FXA-9485
+        // await customs.check(request, email, 'passwordForgotSendOtp');
+
+        request.validateMetricsContext();
+
+        let flowCompleteSignal;
+        if (requestHelper.wantsKeys(request)) {
+          flowCompleteSignal = 'account.signed';
+        } else {
+          flowCompleteSignal = 'account.reset';
+        }
+        request.setMetricsFlowCompleteSignal(flowCompleteSignal);
+
+        const account = await db.accountRecord(email);
+        const code = await otpManager.create(account.uid);
+
+        const ip = request.app.clientAddress;
+        const service = request.payload.service || request.query.service;
+        const { deviceId, flowId, flowBeginTime } = await request.app
```

Comment:
Is there something specific about this ordering that I'm missing here? It looks like there's a chunk of work to get the data needed for metrics with some work done purely to prep the email below that are mingled together. Can the fetching of the account/code/ip/service be put next to the place its used outside of this?

Reply:
No, the order shouldn't matter.  They are all for the email.

> Can the fetching of the account/code/ip/service be put next to the place its used outside of this?

Do you mean inline them into the send email call?

Reply:
No, mainly just having them all right before the email, rather than sprinkled throughout the metrics portion.

Reply:
This is pretty much right before the email though.  There are two more assignment statements for the same purpose after this, then it's the email call.  I'm certain I'm interpreting something quite differently from you but I don't know what. :-/